lib/api.ts

```diff
@@ -99,34 +120,32 @@ function getApi() {
         }
     ];
     let remaining = 0;
-    pending.forEach(function (api) {
+    pending.forEach(api => {
         const isObjCApi = api.module === 'libobjc.A.dylib';
         const functions = api.functions || {};
         const variables = api.variables || {};
         const optionals = api.optionals || {};
 
         remaining += Object.keys(functions).length + Object.keys(variables).length;
 
-        const exportByName = Module
-        .enumerateExportsSync(api.module)
-        .reduce(function (result, exp) {
+        const exportByName = Process
+        .getModuleByName(api.module)
+        .enumerateExports()
+        .reduce((result, exp) => {
             result[exp.name] = exp;
             return result;
-        }, {});
+        }, <any> {});
 
         Object.keys(functions)
-        .forEach(function (name) {
+        .forEach(name => {
             const exp = exportByName[name];
             if (exp !== undefined && exp.type === 'function') {
                 const signature = functions[name];
                 if (typeof signature === 'function') {
                     signature.call(temporaryApi, exp.address);
-                    if (isObjCApi)
-                        signature.call(temporaryApi, exp.address);
                 } else {
-                    temporaryApi[name] = new NativeFunction(exp.address, signature[0], signature[1], defaultInvocationOptions);
-                    if (isObjCApi)
-                        temporaryApi[name] = temporaryApi[name];
+                    const types = <FunctionDeclaration> signature;
```

Comment:
We're currently using `as` casts in existing TS code, so would be good to stick to that.

Reply:
OK, i will change all casts by using `as`.

Reply:
Awesome! Forgot to say that it's preferable to avoid casts when possible, e.g. when the variable's type can be specified and the right-hand side of an assignment can then be type-checked.